### PR TITLE
Port Feeder from Old Chomp

### PR DESCRIPTION
**Pull request by Razgriz1032**
FEEEEEEED.

Made with 4 plastic sheets.

You can make a large reagent container with 5 sheets. Then _somehow_ fill that up and then you can put that into the feeder. Remember to remove the cap off of that large reagent container. Then click and drag on the feeder to someone to begin ~~inflating~~ feeding them.

Ports this thingy
https://github.com/CHOMPstation/CHOMPstation/pull/427

### CODE REVIEW

File: code/game/machinery/feeder.dm [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/obj/machinery/feeder
+	name = "\improper Feeder"
+	icon = 'icons/obj/feeder.dmi'
+	desc = "This is a feeder. Put in a regent container, then click and drag the feeder to someone!"
+	anchored = 0
+	density = 0
```
Comment (Shadow-Quill) [inline code/game/machinery/feeder.dm:6]: 0 to FALSE

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]
Rename the feeder file/feeder DMI to `feeder_ch.dm` and `feeder_ch.dmi`

File: code/game/machinery/feeder.dm [hunk 2]
```diff
@@ -0,0 +1,123 @@
+/obj/machinery/feeder
+	name = "\improper Feeder"
+	icon = 'icons/obj/feeder.dmi'
+	desc = "This is a feeder. Put in a regent container, then click and drag the feeder to someone!"
```
Comment (Shadow-Quill) [inline code/game/machinery/feeder.dm:4]:
```suggestion
	desc = "This is a feeder. Put in a reagent container, then click and drag the feeder to someone!"
```

File: code/game/machinery/feeder.dm [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+/obj/machinery/feeder/var/mob/living/carbon/human/attached = null
+/obj/machinery/feeder/var/obj/item/weapon/reagent_containers/beaker = null
+
+/obj/machinery/feeder/update_icon()
+	if(attached)
+		icon_state = "feeding"
+	else
+		icon_state = ""
+
+	overlays = null
```
Comment (Shadow-Quill) [inline code/game/machinery/feeder.dm:18]: This _should?_ be fine, though usually `overlays.Cut()` is used instead.

File: code/game/machinery/feeder.dm [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+			var/percent = round((reagents.total_volume / beaker.volume) * 100)
+			switch(percent)
+				if(0 to 9)		filling.icon_state = "reagent0"
+				if(10 to 19) 	filling.icon_state = "reagent10"
+				if(20 to 44)	filling.icon_state = "reagent20"
+				if(45 to 59)	filling.icon_state = "reagent45"
+				if(60 to 74)	filling.icon_state = "reagent60"
+				if(75 to 89)	filling.icon_state = "reagent75"
+				if(90 to 94)	filling.icon_state = "reagent90"
+				if(95 to INFINITY)	filling.icon_state = "reagent100"
```
Comment (Shadow-Quill) [inline code/game/machinery/feeder.dm:34]: I'm sure there's a better way to handle this, though I can't puzzle through it. If you'd like to attempt, check out L25 in `code\game\machinery\cell_charger.dm`

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]
few more things I noticed on a second pass

File: vorestation.dme [hunk 5]
```diff
@@ -815,6 +815,7 @@
 #include "code\game\machinery\doorbell_vr.dm"
 #include "code\game\machinery\doppler_array.dm"
 #include "code\game\machinery\exonet_node.dm"
+#include "code\game\machinery\feeder.dm"
```
[the remark this answers is not in the document]
Comment (Razgriz1032) [reply vorestation.dme:818]: Not needed as this is a base and original file.

File: code/game/machinery/feeder.dm [hunk 6]
```diff
@@ -0,0 +1,123 @@
+/obj/machinery/feeder
+	name = "\improper Feeder"
+	icon = 'icons/obj/feeder.dmi'
```
[the remark this answers is not in the document]
Comment (Razgriz1032) [reply code/game/machinery/feeder.dm:3]: Not needed as this is a base and original file.

File: vorestation.dme (hunk 5 above)
[the remark this answers is not in the document]
Comment (Shadow-Quill) [reply vorestation.dme:818]: Other files, such as `casino_ch.dm` and `confetti_grenade_ch.dmi` would disagree, but m'kay.

**Pull request merged**